classes/CartRule.php

```diff
@@ -472,7 +472,7 @@ public static function getCustomerCartRules(
             foreach ($result as $key => $cart_rule) {
                 if ($cart_rule['product_restriction']) {
                     $cr = new CartRule((int) $cart_rule['id_cart_rule']);
-                    $r = $cr->checkProductRestrictionsFromCart(Context::getContext()->cart, false, false);
+                    $r = $cr->checkProductRestrictionsFromCart($cart, false, false);
```

Comment:
To avoid BC, I think something like that is more appropriated ?
```php
$cart = $cart ?? Context::getContext()->cart;
```

Reply:
Why ? `$cart `will exists, as it's the test that scope this line.

Reply:
well, i think it's a bug an the context cart shouldn't be called at all. If you go in private navigation in the backend (without going in the front end) `Context::getContext()->cart` is null.